menuflow/db/room.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, ClassVar, Dict
+
+from asyncpg import Record
+from attr import dataclass
+from mautrix.types import RoomID
+from mautrix.util.async_db import Database
+
+fake_db = Database.create("") if TYPE_CHECKING else None
+
+
+@dataclass
+class Room:
+
+    db: ClassVar[Database] = fake_db
+
+    id: int | None
+    room_id: RoomID
+    variables: Dict | None
+    node_id: str
+    state: str | None
+
+    @classmethod
+    def _from_row(cls, row: Record) -> Room | None:
+        return cls(**row)
+
+    @property
+    def values(self) -> tuple:
+        return (self.room_id, self.variables, self.node_id, self.state)
+
+    async def insert(self) -> str:
+        q = "INSERT INTO room (room_id, variables, node_id, state) VALUES ($1, $2, $3, $4)"
+        await self.db.execute(q, *self.values)
+
+    async def update(self) -> None:
+        q = "UPDATE room SET variables = $2, node_id = $3, state = $4 WHERE room_id = $1"
+        await self.db.execute(q, *self.values)
+
+    @classmethod
+    async def get_by_room_id(cls, room_id: RoomID) -> Room | None:
+        q = f"SELECT id, room_id, variables, node_id, state FROM room WHERE room_id=$1"
```

Comment:
We can use the attribute `columns = "room_id, variables, node_id, state"` for ease of use.

e.g:

`q = f"SELECT id, {columns} FROM room WHERE room_id=$1"`